packages/common/workspace/src/engine/blob.ts

```diff
@@ -1,6 +1,13 @@
 import { DebugLogger } from '@affine/debug';
+import { Slot } from '@blocksuite/global/utils';
 import { difference } from 'lodash-es';
 
+export interface BlobSetEventArgs {
+  key: string;
+  value: Blob;
+  updateShouldProceed: (proceed: boolean) => void;
```

Comment:
Passing a side-effect function in the event doesn't looks very good to me.

Perhaps we can add a `singleBlobSizeLimit` field to limit the maximum size of a single file, and then set `blobEngine.singleBlobSizeLimit = 100mb` in react.

Then add an `onAbortLargeFile` event to trigger the size limit model.

```ts
singleBlobSizeLimit: number = 100mb
onAbortLargeBlob = new Slot<Blob>()
```